### PR TITLE
[CRO] move keys inside initialise function instead of passing

### PR DESCRIPTION
**Pull request by NikitK-deriv**

### CODE REVIEW

**Review by review-deriv — COMMENTED** [review]
PR description is too short. You can use this format:

```
# Description
This is a summary of the changes. This should describe why do we need this change.
Whys and Hows come here.

# Cards
* https://clickup.com/12345
* https://clickup.com/12346

```

File: src/analytics.ts [hunk 1]
```diff
@@ -12,15 +12,19 @@ type Options = {
...
-        if (growthbookKey && growthbookDecryptionKey) {
-            _growthbook = Growthbook.getGrowthBookInstance(growthbookKey, growthbookDecryptionKey, enableDevMode)
-        }
+    const initialise = () => {
+        _rudderstack = RudderStack.getRudderStackInstance(
+            ['.pages.dev', 'git-fork', 'localhost', 'staging'].some(condition =>
+                window.location.hostname.includes(condition)
+            )
+                ? '1lN3tsFD2nruGFgM5F074DC2hMB'
+                : '1oV5agvlcnCZ6IH94wCKM1oR8Pd'
```
Comment (amir-deriv) [inline src/analytics.ts:21]: Why are we hardcoding the rudderstack keys here ?

Comment (NikitK-deriv) [reply src/analytics.ts:21]: discussed in thread: https://deriv-group.slack.com/archives/C05DUEXQLU8/p1701158017753789

File: README.md [hunk 2]
```diff
@@ -93,3 +93,9 @@ If you need to get entire instance directly:
 ```js
 const { ab, tracking } = Analytics?.getInstances()
 ```
+
+If you want to check you ID
```
Comment (amir-deriv) [inline README.md:97]:
```suggestion
If you want to check your ID
```

File: src/analytics.ts [hunk 3]
```diff
@@ -61,6 +60,9 @@ export function createAnalyticsInstance(options?: Options) {
     const setUrl = (href: string) => _growthbook?.setUrl(href)
     const getId = () => _rudderstack?.getUserId() || _rudderstack?.getAnonymousId()
 
+    // for QA testing purposes
+    window.getMyId = () => getId()
+
```
Comment (amir-deriv) [inline src/analytics.ts:65]: do you think we should expose this on production ?

File: src/analytics.ts [hunk 4]
```diff
@@ -5,17 +5,16 @@ import { TCoreAttributes, TEvents } from './types'
...
-    enableDevMode: boolean
+    rudderstackKey?: string
 }
 
 export function createAnalyticsInstance(options?: Options) {
     let _growthbook: Growthbook, _rudderstack: RudderStack
 
-    const initialise = ({ growthbookKey, growthbookDecryptionKey, rudderstackKey, enableDevMode }: Options) => {
+    const initialise = ({ growthbookKey, growthbookDecryptionKey, rudderstackKey }: Options) => {
         _rudderstack = RudderStack.getRudderStackInstance(rudderstackKey)
```
Comment (amir-deriv) [inline src/analytics.ts:15]: if `rudderstackKey` is optional, then someone initialising without rudderstack key breaks the code here, either we should keep rudderstackKey required or if not, then should at least put the condition here to only instantiate rudderstack if we have the key available

The reason is, one could be initialising the growthbook and don't want rudderstack to be initialised, in this case the package will throw the error like `Error: initialising rudderstack with incorrect write key`

Comment (NikitK-deriv) [reply src/analytics.ts:15]: that's correct. after conversation I have to update it

File: src/analytics.ts (hunk 3 above)
Comment (NikitK-deriv) [reply src/analytics.ts:65]: actually we should have this exploit for testing purpose and QA needs

File: src/analytics.ts [hunk 5]
```diff
@@ -61,6 +60,9 @@ export function createAnalyticsInstance(options?: Options) {
     const setUrl = (href: string) => _growthbook?.setUrl(href)
     const getId = () => _rudderstack?.getUserId() || _rudderstack?.getAnonymousId()
 
+    // for QA testing purposes
+    window.getMyId = () => getId()
```
Comment (ali-hosseini-deriv) [inline src/analytics.ts:64]:
```suggestion
    window.getMyId = getId
```
this should do the same.

**Pull request merged**